internal/provider/util.go

```diff
@@ -88,20 +88,34 @@ func readPublicKey(d *schema.ResourceData, prvKey interface{}) error {
 	}
 
 	d.SetId(hashForState(string(pubKeyBytes)))
-	d.Set("public_key_pem", string(pem.EncodeToMemory(pubKeyPemBlock)))
+
+	if err := d.Set("public_key_pem", string(pem.EncodeToMemory(pubKeyPemBlock))); err != nil {
+		return fmt.Errorf("error setting value on key 'public_key_pem': %s", err)
+	}
 
 	// NOTE: ECDSA keys with elliptic curve P-224 are not supported by `x/crypto/ssh`,
 	// so this will return an error: in that case, we set the below fields to emptry strings
 	sshPubKey, err := ssh.NewPublicKey(publicKey(prvKey))
+	var pubKeySSH, pubKeySSHFingerprintMD5, pubKeySSHFingerprintSHA256 string
 	if err == nil {
 		sshPubKeyBytes := ssh.MarshalAuthorizedKey(sshPubKey)
-		d.Set("public_key_openssh", string(sshPubKeyBytes))
-		d.Set("public_key_fingerprint_md5", ssh.FingerprintLegacyMD5(sshPubKey))
-		d.Set("public_key_fingerprint_sha256", ssh.FingerprintSHA256(sshPubKey))
-	} else {
-		d.Set("public_key_openssh", "")
-		d.Set("public_key_fingerprint_md5", "")
-		d.Set("public_key_fingerprint_sha256", "")
+
+		pubKeySSH = string(sshPubKeyBytes)
+		pubKeySSHFingerprintMD5 = ssh.FingerprintLegacyMD5(sshPubKey)
+		pubKeySSHFingerprintSHA256 = ssh.FingerprintSHA256(sshPubKey)
+	}
+
+	if err := d.Set("public_key_openssh", pubKeySSH); err != nil {
```

Comment:
In the original `else` block, all values are set to empty strings. Is that behavior still needed? (my guess is no)

Reply:
So, I wanted to avoid having to write the same `if err := d.Set...` 6 times, was kinda ugly. And the static analysis reported that we were not checking the return error from `.Set`.

So, I have initialized `publicKeySSH*` variables to empty strings: if they get a value (i.e. `ssh.NewPublicKey()` does not error), we set those strings to a proper value. Otherwise, they are already empty string.

This preserves the previous behaviour WHILE making sure we call `.Set` for each attribute only once.

Reply:
This looks good -- its important to preserve the `""` setting behavior where it currently exists since not doing so is an odd backwards compatibility issue until a next major version (which we may want to consider breaking).